solr/core/src/java/org/apache/solr/core/ConfigOverlay.java

```diff
@@ -34,15 +34,15 @@
  * performed on tbhis gives a new copy of the object with the changed value
  */
 public class ConfigOverlay implements MapSerializable {
-  private final int znodeVersion;
```

Comment:
is there any reason why this rename was necessary ?

Reply:
Because its no longer necessarily the znode version, it could be the file "version" for a file-system based configSet.

Reply:
If a znode with version 0 is deleted and recreated quickly then it comes back with the same version and this missed updates before